deployment/cdk/opensearch-service-migration/lib/common-utilities.ts

```diff
@@ -187,6 +187,31 @@ export function createDefaultECSTaskRole(scope: Construct, serviceName: string):
     return serviceTaskRole
 }
 
+export function createSnapshotOnAOSRole(scope: Construct, artifactS3Arn: string, migrationConsoleTaskRoleArn: string): Role {
+    const snapshotRole = new Role(scope, `SnapshotRole`, {
+        assumedBy: new ServicePrincipal('es.amazonaws.com'),
+        description: 'Role that grants OpenSearch Service permissions to access S3 to create snapshots',
```

Comment:
Can we add a deterministic name to avoid suffix? `OSMigrations-new-cdk-us-east-2-SnapshotRole53D7C789-18fLjvR4jHRc`?